language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/rename/RenameUtil.java

```diff
@@ -0,0 +1,136 @@
+*  under the License.
+*/
+package org.ballerinalang.langserver.rename;
+
+import org.ballerinalang.langserver.common.utils.CommonUtil;
+import org.ballerinalang.langserver.workspace.WorkspaceDocumentManager;
+import org.eclipse.lsp4j.Location;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.RenameParams;
+import org.eclipse.lsp4j.TextDocumentEdit;
+import org.eclipse.lsp4j.TextEdit;
+import org.eclipse.lsp4j.VersionedTextDocumentIdentifier;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Utility methods for rename operation.
+ */
+public class RenameUtil {
+
+    /**
+     * Get the list of rename related TextEdits.
+     * @param locationList          List of locations of occurrences
+     * @param documentManager       {@link WorkspaceDocumentManager} instance
+     * @param newName               New name to be replaced with
+     * @param replaceSymbolName     Symbol name being replaced
+     * @return {@link List}         List of TextEdits
+     */
+    public static List<TextDocumentEdit> getRenameTextEdits(List<Location> locationList,
+                                                            WorkspaceDocumentManager documentManager, String newName,
+                                                            String replaceSymbolName) {
+        Map<String, ArrayList<Location>> documentLocationMap = new HashMap<>();
+        List<TextDocumentEdit> documentEdits = new ArrayList<>();
+        Comparator<Location> locationComparator = (location1, location2)
+                ->location1.getRange().getStart().getCharacter() - location2.getRange().getStart().getCharacter();
+        
+        locationList.forEach(location -> {
+            if (documentLocationMap.containsKey(location.getUri())) {
+                documentLocationMap.get(location.getUri()).add(location);
+            } else {
+                ArrayList<Location> locations = new ArrayList<>();
+                locations.add(location);
+                documentLocationMap.put(location.getUri(), locations);
+            }
+        });
+        
+        documentLocationMap.forEach((uri, locations) -> {
+            Collections.sort(locations, locationComparator);
+            String fileContent = documentManager.getFileContent(CommonUtil.getPath(uri));
+            String[] contentComponents = fileContent.split("\\n|\\r\\n|\\r");
+            int lastNewLineCharIndex = Math.max(fileContent.lastIndexOf("\n"), fileContent.lastIndexOf("\r"));
+            int lastCharCol = fileContent.substring(lastNewLineCharIndex + 1).length();
+
+            for (Location location : locations) {
+                int line = location.getRange().getStart().getLine();
+                StringBuilder lineComponent = new StringBuilder(contentComponents[line]);
```

Comment:
Since we use a number of alterations over the lineComponent and to avoid creating String objects since String is immutable, used the StringBuilder instead